index.js

```diff
@@ -10,125 +10,127 @@ import {
-  filteredTodos = filteredTasks.length
-    ? filteredTasks.filter((todo) => todo.isDone === isDone)
-    : todos.filter((todo) => todo.isDone === isDone);
+const handleLoadMoreTask = () => {
+  if (todos && currentPage < totalPageCount) {
+    currentPage++;
+  }
 
-  filteredTodos.map((todo) => {
-    const newTask = createTaskElement(todo);
-    taskList.insertBefore(newTask, taskList.children[1]);
-  });
+  renderTodoList(todos);
 };
 
-export const keepCard1Elements = () => {
-  const elementsToKeep = [];
+export const renderLoadMoreButton = (todos) => {
+  if (todos.length >= TASK_PER_PAGE && currentPage !== totalPageCount) {
+    loadMoreButton.classList.remove("hide"),
+      loadMoreButton.classList.add("show");
 
-  for (let i = 0; i < taskList.children.length; i++) {
-    if (taskList.children[i].id === "card1") {
-      elementsToKeep.push(taskList.children[i]);
-    }
+    return;
   }
 
-  taskList.innerHTML = "";
+  loadMoreButton.classList.remove("show"), loadMoreButton.classList.add("hide");
 
-  elementsToKeep.map((element) => {
-    taskList.appendChild(element);
-  });
+  return;
 };
 
-const debounce = (handleSearch, delay) => {
-  let timer;
+const todoStateFilter = (status = ALL, todos) => {
+  if (status === ALL) {
+    return todos;
+  } else if (status === COMPLETE) {
+    return todos.filter((todo) => todo.isDone === true);
+  }
+  return todos.filter((todo) => todo.isDone === false);
+};
 
-  return function (...args) {
-    clearTimeout(timer);
+const searchFilter = (searchText, todos) => {
+  if (searchText !== "") {
+    return todos.filter((task) =>
+      task.value.toLowerCase().includes(searchText)
+    );
+  }
+  return todos;
+};
 
-    timer = setTimeout(() => {
-      handleSearch(...args);
-    }, delay);
+const debounce = (searchFilter, delay) => {
+  let timer;
+  return function () {
+    clearTimeout(timer);
+    return new Promise((resolve) => {
+      timer = setTimeout(() => {
+        const result = searchFilter.apply(this, arguments);
+        resolve(result);
+      }, delay);
+    });
   };
 };
 
-const handleSearch = () => {
-  keepCard1Elements();
-
-  const searchText = searchInput.value.toLowerCase().trim();
+if (todos.length === 0) {
+  disableFilterBar();
+}
 
-  filteredTasks = filteredTodos?.length
-    ? filteredTodos.filter((task) =>
-        task.value.toLowerCase().includes(searchText)
-      )
-    : todos.filter((task) => task.value.toLowerCase().includes(searchText));
+export const renderTodoList = (todos) => {
+  let auxilaryTodos = [];
+  auxilaryTodos = [...searchFilter(searchText, todos || [])];
+  auxilaryTodos = [...todoStateFilter(filterStatus, auxilaryTodos)];
+  auxilaryTodos = [...paginationFilter(auxilaryTodos)];
+  taskList.innerHTML = "";
+  taskList.appendChild(taskInputCard);
 
-  filteredTasks.map((task) => {
-    const newTask = createTaskElement(task);
-    taskList.insertBefore(newTask, taskList.children[1]);
+  auxilaryTodos.map((todo) => {
```

Comment:
Do we need `map` here?

Reply:
We can use forEach here, as we only appending to DOM